src/Vector3.h

```diff
@@ -235,6 +235,12 @@ class Vector3
 		return *this;
 	}
 
+	/** Provides a hash of a vector's contents */
+	size_t operator()(const Vector3<T> & a_Vector) const
+	{
+		return ((std::hash<T>()(a_Vector.x) ^ (std::hash<T>()(a_Vector.y) << 1)) ^ std::hash<T>()(a_Vector.z));
```

Comment:
Reasoning about the entropy and trying to force a collision. Within a chunk, the four bits of entropy in each of the X and Z direction is concentrated in the lower order bits. You squash all that entropy into 5 bits.